test/test_framework.py

```diff
@@ -461,11 +462,18 @@ def on_foo(self, event):
                 self.seen.append(f"on_foo:{event.handle.kind}")
                 event.defer()
 
+            def on_bar(self, event):
+                self.seen.append(f"on_bar:{event.handle.kind}")
+
         pub = MyNotifier(framework, "1")
         obs = MyObserver(framework, "1")
 
-        framework.observe(pub.on.foo, obs)
+        # Confirm that temporary persistence of BoundEvents doesn't cause errors,
+        # and that events can be observed.
+        for bound_event in [pub.on.foo, pub.on.bar]:
```

Comment:
So this would trigger the issue because the '.on' is accessed 2 times?

Reply:
Each access would result in `type(self)(emitter)` execution which would trigger the issue because there would be 2 instances present for the lifetime of the loop.